src/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -30,11 +30,6 @@ internal static async Task<int[]> HandleRequestHelperAsync(
             RequestContext context,
             CancellationToken cancellationToken)
         {
-            if (ranges.Length == 0)
-            {
-                return [];
-            }
```

Comment:
i actually like this.  i would just make `ranges` nullable and say that if it is null that you get the entire doc. but if it is empty you get nothing back.

Reply:
note: this was not a joke :)